src/condition/scalar_condition.cpp

```diff
@@ -60,8 +60,12 @@ std::optional<condition_match> eval_object(Iterator &it, std::string_view addres
 
                 DDWAF_TRACE("Target {} matched parameter value {}", address, highlight);
 
-                return {{{{"input"sv, object_to_string(dst), address, it.get_current_path()}},
-                    {std::move(highlight)}, matcher.name(), matcher.to_string(), ephemeral}};
+                if constexpr (std::is_same_v<ResultType, bool>) {
+                    return true;
+                } else {
+                    return {{{{"input"sv, object_to_string(dst), address, it.get_current_path()}},
```

Comment:
> difficult to follow. You clearly expect a certain type here, judging by the very specific constructor but it's not mentioned until later in the file. Maybe adding requires(std::is_same_v<ResultType, bool> || std::is_same_v<ResultType, std::optional<condition_match>>) .

Alright that's fair, I'll add the requires.

> Although, I'd very much prefer you don't have this where you build the condition_match in this template for the scalar_condition and for the negated condition you build it in the caller... seems that what you need is a boolean to indicate to this template if you should generate the match when res is true or false.

I'm not sure that would work because on the negated scalar condition, the match is only generated if there is no match when calling `eval_target` / `eval_object`. So in principle it could be generated in `eval_target` but not in `eval_object`. Conversely, the scalar condition can only generate the match on `eval_object`, because `eval_target` already lacks sufficient context.

Reply:
In saying that, I should be able to move the match generation to `eval_target` and just propagate the needed context from `eval_object` in the scalar condition case.

Reply:
The code ends up kinda ugly if I split things, so I added the requires.